src/units/unit.cpp

```diff
@@ -888,19 +888,39 @@ void unit::generate_traits(bool must_have_only)
 	random_traits_ = false;
 }
 
-std::vector<std::string> unit::get_traits_list() const
+std::vector<std::string> unit::get_modifications_list(const std::string& mod_type) const
 {
 	std::vector<std::string> res;
 
-	for(const config& mod : modifications_.child_range("trait"))
-	{
-			// Make sure to return empty id trait strings as otherwise
-			// names will not match in length (Bug #21967)
+	for(const config& mod : modifications_.child_range(mod_type)){
+		// Make sure to return empty id trait strings as otherwise
+		// names will not match in length (Bug #21967)
+		res.push_back(mod["id"]);
+	}
+	if(mod_type == "advancement"){
+		for(const config& mod : modifications_.child_range("advance")){
 			res.push_back(mod["id"]);
+		}
 	}
 	return res;
 }
 
+std::size_t unit::modification_count(const std::string& type) const
+{
+	//return numbers of modifications of same type, same without ID.
+	std::size_t res = 0;
+	for(const config& item : modifications_.child_range(type)) {
+		if(!item.empty()){
```

Comment:
What is this check for?

Reply:
if i don't use if loop i had an warning what variable item is not used during compilation.

Reply:
changed for direct checking of modifications_.child_range(type) size